dialogue-core/src/test/resources/tracing/DialogueChannelTest/traces_on_retries.log

```diff
@@ -1,30 +1,31 @@
+{"traceId":"3519228122fee79e","parentSpanId":"2a5d7b91bd74a924","spanId":"ef71041028f73044","type":"LOCAL","operation":"Dialogue-request-attempt initial","startTimeMicroSeconds":1619611357528152,"durationNanoSeconds":971693,"metadata":{}}
+{"traceId":"3519228122fee79e","parentSpanId":"d7836b17cef5db99","spanId":"866a99f10ab0c9b9","type":"LOCAL","operation":"dialogue-RetryingChannel-scheduler","startTimeMicroSeconds":1619611357528000,"durationNanoSeconds":1993389,"metadata":{}}
+{"traceId":"3519228122fee79e","parentSpanId":"257e95ea327f70b4","spanId":"896d67d85b76c8b3","type":"LOCAL","operation":"retry-backoff","startTimeMicroSeconds":1619611357529879,"durationNanoSeconds":452614274,"metadata":{"serviceName":"service","endpointName":"endpoint","failures":"4","channel":"my-channel"}}
+{"traceId":"3519228122fee79e","parentSpanId":"fc414565b6a32ddd","spanId":"5f60a24389d47a72","type":"LOCAL","operation":"Dialogue-http-request","startTimeMicroSeconds":1619611357982728,"durationNanoSeconds":576158,"metadata":{"channel":"my-channel","mesh":"false","outcome":"failure","http.status_code":"429"}}
+{"traceId":"3519228122fee79e","parentSpanId":"5f60a24389d47a72","spanId":"1cc36e6dbea1ac05","type":"CLIENT_OUTGOING","operation":"Dialogue-http-request initial","startTimeMicroSeconds":1619611357982799,"durationNanoSeconds":737015,"metadata":{}}
+{"traceId":"3519228122fee79e","parentSpanId":"cff8c71cc0a4de65","spanId":"b9eede02cfc9a230","type":"LOCAL","operation":"Dialogue-request-attempt","startTimeMicroSeconds":1619611357982587,"durationNanoSeconds":1353657,"metadata":{"endpointService":"service","endpointName":"endpoint","http.method":"POST","channel":"my-channel","mesh":"false","outcome":"failure","http.status_code":"429"}}
+{"traceId":"3519228122fee79e","parentSpanId":"b9eede02cfc9a230","spanId":"fc414565b6a32ddd","type":"LOCAL","operation":"Dialogue-request-attempt initial","startTimeMicroSeconds":1619611357982666,"durationNanoSeconds":1428971,"metadata":{}}
+{"traceId":"3519228122fee79e","parentSpanId":"76a3f96c2bd32ebb","spanId":"257e95ea327f70b4","type":"LOCAL","operation":"Dialogue-RetryingChannel","startTimeMicroSeconds":1619611357020816,"durationNanoSeconds":964010756,"metadata":{"serviceName":"service","endpointName":"endpoint","failures":"5","channel":"my-channel"}}
+{"traceId":"3519228122fee79e","parentSpanId":null,"spanId":"4791d408917316f4","type":"LOCAL","operation":"Dialogue: request service#endpoint","startTimeMicroSeconds":1619611357020205,"durationNanoSeconds":970171438,"metadata":{"endpointService":"service","endpointName":"endpoint","http.method":"POST","channel":"my-channel","mesh":"false","outcome":"failure","http.status_code":"429"}}
+{"traceId":"3519228122fee79e","parentSpanId":"866a99f10ab0c9b9","spanId":"cff8c71cc0a4de65","type":"LOCAL","operation":"dialogue-RetryingChannel-scheduler","startTimeMicroSeconds":1619611357982457,"durationNanoSeconds":8481191,"metadata":{}}
```

Comment:
Thoughts on what happened?

Reply:
Feels like this should've been there from the start?

Reply:
My guess is that the scheduler runnable is still executing while the test completes, which allows the test to record spans up to, but not including the final `dialogue-RetryingChannel-scheduler`.
In both before and after, we have the correct number of `Dialogue-request-attempt` and `Dialogue-request-attempt initial` spans.